packages/@o3r/rules-engine/src/engine/ruleset-executor.ts

```diff
@@ -267,7 +267,15 @@ export class RulesetExecutor {
               if (this.rulesEngine.debugMode) {
                 output.evaluation = handleRuleEvaluationDebug({ ...rule, inputFacts }, this.ruleset.name, output.actions, output.error, runtimeFactValues, factValues, oldFactValues);
               } else if (output.error) {
-                this.rulesEngine.logger?.error(output.error);
+                let errorMsg;
+                if (output.error instanceof Error) {
+                  errorMsg = output.error.toString();
+                } else if (typeof output.error === 'string') {
+                  errorMsg = output.error;
```

Comment:
```suggestion
                if (output.error instanceof Error || typeof output.error === 'string') {
                  errorMsg = output.error.toString();
```

Reply:
maybe we could export this logic in a function like
```typescript
errorMsg = isJsonSerializable(output.error) ? JSON.stringify(output.error) : output.error.toString();
```